### PR TITLE
Shaunna Wiens --carets

### PR DESCRIPTION
**Pull request by skwiens**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I mostly looked at the documents as well as played with Postman to look at the results. |
| Describe your API Wrapper.  How did you decide on the methods you created? | My API wrapper includes two methods, both of which send an API request to Edamam API. The search method uses a q search to search for a term. The item_search method uses an r search to search for the item.  Another possible method would be to store the recipes searched by creating a class variable and then using r to search through there. The benefit to this would be that it would require less requests to the API. This could be beneficial if I ever wanted more than one person to use my site because I already get too many emails from Edamam saying that I'm going past their 5 searches per minute rule.  |
| Describe an edge case or failure case test you wrote for your API Wrapper. | An edge case that I wrote for the API Wrapper is the case in which a bogus ID and Key are used. Currently the largest bug that I haven't figured out how to test or fix in my program is that if I type into the url bar a random uri (like "bogus") it returns a route not found error.  |
| Explain how VCR aids in testing an API. |  VCR aids in testing because it has bootlegging abilities. It allows the user to record results from queries. This is helpful because Edamam frequently sends me helpful emails telling me I've passed my limit...this would most certainly be true if I ran my tests which run in under a minute. I'm also thinking creating my own 'version' by recording results in models in the database and checking those before a query but might not have time. |
| What is the Heroku URL of your deployed application? | https://munchers.herokuapp.com - I deployed early enough that I got the munchers name for heroku!!!! :)  |

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  APP_KEY = ENV["EDAMAM_KEY"]
+
+  def self.item_search(item_uri, app_id=nil, app_key=nil)
+    app_id ||= APP_ID
+    app_key ||= APP_KEY
+
+    url = BASE_URL + "r=#{item_uri}" "&app_id=#{app_id}" + "&app_key=#{app_key}"
+
+    data = parsed_response(HTTParty.get(url))
+    return nil if data.nil?
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:16]: You probably also need to have 
`return nil if data[0].nil?`

Because that's what happens with a bad URI.  It's causing your app to crash for the bogus URI problem.

File: test/controllers/recipes_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+require "test_helper"
+
+describe RecipesController do
+  describe "home" do
+    it "successfully opens" do
+      VCR.use_cassette("home") do
+        get root_path
+        must_respond_with :success
+        flash[:status].must_equal :success
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:9]: This seems wrong

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good number of commit  |
| Comprehension questions	| Bootlegging abilities!  Instead of using a class variable, you would instead need to "cache" the recipes in the database with Active Record because otherwise you can't remember data between requests.  Congrats on the munchers name!   |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | Good semantic HTML |
| Errors are reported to the user | Well done |
| API Wrapper to handle the API requests | Check, I did leave one note of an issue with the wrapper |
| Controller testing | Check, but I left a note in your code. |
| Lib testing | Check  |
| Search Functionality | Check |
| List Functionality |Check  |
| Show individual item functionality (link to original recipe opens in new tab) | Doesn't open in a new tab, but the link's there.  |
| **Styling** |  |
| Foundation Styling for responsive layout | Nicely responsive |
| List View shows 10 items at a time/pagination | Check |
| The app is styled to create an attractive user interface | Good although the pagination could be better styled. |
| **API Features** | |
| The App attributes Edaman | It doesn't link back to Edamam |
| The VCR casettes do not contain the API key |  |
| **External Resources** |  |
| Link to deployed app on Heroku | Early Deployment, well done! |
|  **Overall** | This was a solid submission.  I left a few comments in your code, but this was well done. |
